### DISCUSSION TITLE
How to install third-party services at install time

### DISCUSSION THREAD

**Discussion by zzkkui** [tauri-apps/tauri#6436] [Q&A]
If my service depends on a third-party service (for example, an exe file needs to be installed), how to include the third-party service when packaging. And automatically install this 3rd party service during installation? I tried bundler.externalBin , but how do I get it to install automatically during installation?

**Comment by ProfNuru** [comment]
Have you gotten a solution to this?

**Comment by CaseSens** [comment]
the `bundler.externalBin` didn't work? You should be able to [run the binary (exe) from Rust](https://v2.tauri.app/develop/sidecar/#running-it-from-rust) using this.

If somehow that doesn't work as intended, you can always, on launch, run the .exe manually using `(std::process::Command)` .
This could look something like:

```rust
fn main() {
  #[cfg(target_os = "windows")]
  {
    let _ = Command::new("<path\to\your\executable.exe")
      .spawn()
      .expect("Failed to start executable");
  }

  tauri::Builder::default()
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```
